### ISSUE TITLE
[Feature request] Users' collections

### ISSUE THREAD

**Issue by Karmalakas** [diaspora/diaspora#7903]
Tried searching, but didn't find any collections request. I'm trying to get used to D* after G+ :) And the main feature I'm missing, is G+ style Collections. It also can be achieved through hashtags, but then it would need to change the following system (adding tags/persons excludes).

What I'd want, is to follow a specific tag from specific people or even one person. Also sometimes I'd like to follow a person except posts with specific tag from that person. There's no way to achieve this now. G+ Collections let me do exactly that, by following some collection or by following a person and excluding any of his/her collections.

**Comment by goobertron** [comment]
I see this as a sub-set of [this discussion on expanding the functionality of tags](https://discourse.diasporafoundation.org/t/improving-and-expanding-hashtags-usability/213).

There is also a specific discussion about [adding a G+-style 'collection' feature](https://discourse.diasporafoundation.org/t/collections-digest-feature/1898).

Personally I think the best way to solve this is to use #tags and allow filtering of various sorts. This fits in with diaspora\*'s existing ecosystem. Restricting results to a particular user (or eliminating a particular user) could be one of the ways of filtering tags.

Anyway, please head over to one of those discussions if you wish to discuss this.

**Comment by Flaburgan** [comment]
>What I'd want, is to follow a specific tag only from specific people or even one person

This is **exactly** what I have in mind for years. I felt that https://github.com/diaspora/diaspora/issues/2019#issuecomment-14488582 was enough to track this issue but maybe it's not. Anyway I have mockups in mind etc, I need to spend some time to draft that because this is really what I miss the most in diaspora*.

**Comment by Karmalakas** [comment]
@Flaburgan, I'm talking not only about search, but also somehow to set these parameters, so matching posts would appear on my stream.

**Comment by goobertron** [comment]
This is far more closely related to #5221; in fact, it is actually a subset of the proposal in that issue, which includes the following:

> We have to define elements that can be used in the filter (inspired by @Alienglyph-s post)
a) #tagThatIsInsideThePost
b) **senderWhoWroteThePost**
c) **commenterWhoAddedCommentToThePost**
d) **@somoneBeeingReferenced**
e) someTextInThePost
f) ageOfThePost (i.e. not older than one week)

b, c and d all relate to filtering by user name.

**Comment by Flaburgan** [comment]
>I'm talking not only about search

Yeah, my idea was to allow users to create custom streams based on pinned advanced searches. I really need to work on that topic.

**Comment by HankG** [comment]
> Yeah, my idea was to allow users to create custom streams based on pinned advanced searches

I loved that idea...

**Comment by stragu** [comment]
I really like this. It's slightly related to what I mentioned in issue https://github.com/diaspora/diaspora/issues/7759 : **filtering the user's own posts out**.

Also, if this custom stream can **include or exclude existing aspects**, it would be wonderful. The nature of Diaspora means that I ended up following lots of people who I don't know personally, but now that I have a few friends and family members of there, I would like to easily have access to a stream that only includes the aspects "Family" and "Friends" (and excludes my own posts), as those posts get easily flooded out by the rest of the community.

**Comment by stragu** [comment]
I really like this. It's slightly related to what I mentioned in issue https://github.com/diaspora/diaspora/issues/7759 : **filtering the user's own posts out**.

Also, if this custom stream can **include or exclude existing aspects**, it would be wonderful. The nature of Diaspora means that I ended up following lots of people who I don't know personally, but now that I have a few friends and family members in my contacts, I would like to easily have access to a stream that only includes the aspects "Family" and "Friends" (and excludes my own posts), as those posts get easily flooded out by the rest of the community.

**Comment by goobertron** [comment]
Please could we move discussion about ideas for what this feature  might look like to the [Discourse discussion for the topic](https://discourse.diasporafoundation.org/t/collections-digest-feature/1898/)? This GitHub issue can be reserved for actual technical discussion further down the line.
